video-library-adapter/xlsx-adapter/src/main/java/it/proconsole/library/video/adapter/xlsx/repository/workbook/FilmWorkbookRepository.java

```diff
@@ -0,0 +1,112 @@
+package it.proconsole.library.video.adapter.xlsx.repository.workbook;
+
+import it.proconsole.library.video.adapter.xlsx.exception.InvalidXlsxFileException;
+import it.proconsole.library.video.adapter.xlsx.model.FilmReviewRow;
+import it.proconsole.library.video.adapter.xlsx.model.FilmRow;
+import org.apache.poi.openxml4j.exceptions.InvalidOperationException;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.DateUtil;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.lang.Nullable;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+
+public class FilmWorkbookRepository {
+  private static final int FILM_SHEET = 0;
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
+  private final String xlsxPath;
+
+  private long filmReviewId = 1L;
+
+  public FilmWorkbookRepository(String xlsxPath) {
+    this.xlsxPath = xlsxPath;
+  }
+
+  public List<FilmRow> findAll() {
+    filmReviewId = 1L;
+    try {
+      var sheets = new XSSFWorkbook(xlsxPath);
+      var filmsSheet = sheets.getSheetAt(FILM_SHEET);
+      var spliterator = filmsSheet.spliterator();
+      var films = StreamSupport.stream(spliterator, false)
+              .skip(3)
+              .filter(this::emptyRows)
+              .map(this::adaptRow)
+              .toList();
+      sheets.close();
```

Comment:
finally?

Reply:
Cannot call finally on an object created within try, used instead the Closeable feature